packages/fe-fpm-writer/src/building-block/types.ts

```diff
@@ -0,0 +1,342 @@
+/**
+ * Building block type.
+ *
+ * @enum {string}
+ */
+export enum BuildingBlockType {
+    FilterBar = 'filter-bar',
+    Chart = 'chart',
+    Field = 'field'
+}
+
+/**
+ * Represents a building block control.
+ */
+export interface BuildingBlock {
+    /**
+     * Building block type.
+     */
+    buildingBlockType: BuildingBlockType;
+
+    /**
+     * Defines the path of the context used in the current page or block. This setting is defined by the framework.
+     */
+    contextPath?: string;
+
+    /**
+     * ID of the building block.
+     */
+    id: string;
+
+    /**
+     * Defines the relative path of the property in the metamodel, based on the current contextPath.
+     */
+    metaPath?: string;
+}
+
+/**
+ * Represents a Chart building block control.
+ * Usually, a contextPath and metaPath is expected.
+ *
+ * @example
+ * <macro:Chart id="Mychart" contextPath="/RootEntity" metaPath="@com.sap.vocabularies.UI.v1.Chart" />
+ * @extends {BuildingBlock}
+ */
+export interface Chart extends BuildingBlock {
```

Comment:
Could you please also add the optional event handlers.

Reply:
@tobiasqueck: I see that I missed the `selectionChange` event. I will add this. I haven't added any inherited events, properties or methods. Let me know if you need any other events or properties and I will add them for now. Hopefully, when the types from Fiori elements are used, all properties can be used.

Reply:
I added the `selectionChange` event to the interface and the template.

Reply:
Thanks, I am ok with it for now, and we can enhance it later.